frontend/src/app/main/data/workspace/libraries.cljs

```diff
@@ -139,6 +139,26 @@
                (sync-file (:current-file-id state) file-id)
                (dwu/commit-undo-transaction))))))
 
+(defn rename-color
+  [file-id id new-name]
+  (us/assert ::us/uuid file-id)
+  (us/assert ::us/uuid id)
+  (us/assert ::us/string new-name)
+  (ptk/reify ::rename-color
+    ptk/WatchEvent
+    (watch [it state _]
+      (let [data        (get state :workspace-data)
+            [path name] (cph/parse-path-name new-name)
+            object      (get-in data [:colors id])
+            new-object  (assoc object :path path :name name)
+            changes     (-> (pcb/empty-changes it)
+                            (pcb/with-library-data data)
+                            (pcb/update-color new-object))]
+        (rx/of (dwu/start-undo-transaction)
+               (dch/commit-changes changes)
+               (sync-file (:current-file-id state) file-id)
+               (dwu/commit-undo-transaction))))))
+
```

Comment:
Why don't use `update-color`, as it's used in `main/ui/workspace/sidebar/assets.cljs` to rename colors?

Reply:
fixed